### DISCUSSION TITLE
GEO & MEO/LEO satellite alignment calculation

### DISCUSSION THREAD

**Discussion by philsellick** [skyfielders/python-skyfield#810] [Q&A]
Hi 

I am trying to figure out if it is possible or of anyone has tried to derive the Time when a LEO/MEO satellite is directly in front of a GEO satellite from a specific lat/long position on earth (ie a Ground Station).  We ar6e looking to predict when a ground station would need to stop transmission to a LEO/MEO satellite to avoid interference with the GEO one. I have the TLE's for both, but it would seem a very iterative process to derive the time when the LEO/MEO Azimuth/Elevation == GEO Azimuth/Elevation from a specific locaiton , wondering if there is quicker way than parsing through tons of data :-)

**Comment by brandon-rhodes** [comment]
Skyfield has a [find_minima()](https://rhodesmill.org/skyfield/api.html#skyfield.searchlib.find_minima) routine that might help you. If you graph the `.separation_from()` angle between the two satellite positions, how often do minima occur? If they were, say, every 90 minutes, you could probably set `find_minima()` searching by providing it with data points maybe 30 minutes or 20 minutes apart—which would let you avoid, say, having to compute each satelllite's position for every second of the day to look for the moments when they're closest in the sky.

It would probably also save time to not bother to compute alt/az, but instead just watch for when the raw position vectors are closest.

**Reply by sellick-phil** [reply]
Brilliant thanks the below works, with some caveats, which we will work on, but will share here anyway should anyone want it. The below is a bit long winded but we are just experimenting for now . Using numba doesn't speed it up much as we use pandas not np for the dataframe analysis. We are really only interested if there is a potential conjunction, so we just check the angle of separation, for each min and then log the times, for analysis. I think however it is also showing angles through the earth as well (i.e. angle of separation but the opposite side) , as from London we see conjunctions from Geo Sats in Australia :-) . We will keep on it , but thanks very much  again, very useful and I suspect and increasingly valuable thing that will be needed by operators, certainly regulators in the UK are asking for avoidance plans for TX now with GEO etc 

for meo in meoSatTLE:
        for geo in geoSatTLE:
            meoSat = meoSatTLE[meo]
            geoSat = geoSatTLE[geo]
            for day in range (1,delta.days):
                for hour in range(1,24):
                    for sat_min in range(1,60):
                        t = ts.utc(d1.year, d1.month, day, hour, sat_min)
                        usno = api.Topos(lat, long, elevation_m=elv)
                        pos1 = (meoSat - usno).at(t)
                        pos2 = (geoSat - usno).at(t)
                        AOS = pos1.separation_from(pos2)
                        if AOS.degrees <= conjunctionAngle:
                            timeStamp=d1+timedelta(days=day, hours=hour, minutes=sat_min)

**Comment by philsellick** [comment]
Ok thanks I will look at it . It is more to calculate the angle from the
ground station so if the MEO satellite comes Infront of the GEO need to
know specifically what times that happens. The beam angles are narrow so if
the look angles are more than say 5' apart in both el & azm looking from
the GS then thats fine

Will dig some more, I think there will be an increasing need to do this and
coordinate between LEO/MEO & GEO and to mute and unmute as per convention

Phil Sellick.
@psellick
Phone : +44 7810 800 058
